src/backend/allocator/dmabuf.rs

```diff
@@ -0,0 +1,177 @@
+//! Module for [dmabuf](https://01.org/linuxgraphics/gfx-docs/drm/driver-api/dma-buf.html) buffers.
+
+use super::{Buffer, Format, Modifier};
+use std::os::unix::io::RawFd;
+use std::sync::{Arc, Weak};
+
+const MAX_PLANES: usize = 4;
+
+#[derive(Debug)]
+pub(crate) struct DmabufInternal {
+    pub num_planes: usize,
+    pub offsets: [u32; MAX_PLANES],
+    pub strides: [u32; MAX_PLANES],
+    pub fds: [RawFd; MAX_PLANES],
+    pub width: u32,
+    pub height: u32,
+    pub format: Format,
+}
+
+#[derive(Debug, Clone)]
+/// Strong reference to a dmabuf handle
+pub struct Dmabuf(pub(crate) Arc<DmabufInternal>);
+
+#[derive(Debug, Clone)]
+/// Weak reference to a dmabuf handle
+pub struct WeakDmabuf(pub(crate) Weak<DmabufInternal>);
+
+impl PartialEq for Dmabuf {
+    fn eq(&self, other: &Self) -> bool {
+        Arc::ptr_eq(&self.0, &other.0)
+    }
+}
+impl Eq for Dmabuf {}
+
+impl PartialEq<WeakDmabuf> for Dmabuf {
+    fn eq(&self, other: &WeakDmabuf) -> bool {
+        if let Some(dmabuf) = other.upgrade() {
+            return Arc::ptr_eq(&self.0, &dmabuf.0);
+        }
+        false
+    }
+}
+
+impl PartialEq for WeakDmabuf {
+    fn eq(&self, other: &Self) -> bool {
+        if let Some(dmabuf) = self.upgrade() {
+            return &dmabuf == other;
+        }
+        false
+    }
+}
+
+impl Buffer for Dmabuf {
+    fn width(&self) -> u32 {
+        self.0.width
+    }
+
+    fn height(&self) -> u32 {
+        self.0.height
+    }
+
+    fn format(&self) -> Format {
+        self.0.format
+    }
+}
+
+impl Dmabuf {
+    pub(crate) fn new(
+        src: &impl Buffer,
```

Comment:
Could be worth adding a comment explaining that the `src` buffer will only be used as a reference for buffer size & format, but not its actual content.